sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -1361,6 +1416,358 @@ __ESIMD_API void lsc_store2d(T *Ptr, unsigned SurfaceWidth,
       Vals.data());
 }
 
+/// <summary>
+///  Container class to hold parameters for \c load2d/store2d \c functions
+/// </summary>
+/// @tparam T Type of data to load/store
+/// @tparam BlockWidth the block width in number of elements
+/// @tparam BlockHeight block height in number of elements
+/// @tparam NBlocks Number of blocks
+template <typename T, int BlockWidth, int BlockHeight, int NBlocks>
+class config2d_mem_access {
```

Comment:
Ops, I missed this class. I believe the idea was to have _2d in the name. I.e. config_2d_mem_access, and have it consistent with lsc_load_2d()